launch_testing/launch_testing/__init__.py

```diff
@@ -40,17 +42,22 @@ class InMemoryHandler(LineOutput):
 
     def __init__(
         self, name, launch_descriptor, expected_lines, regex_match=False,
-        filtered_prefixes=None, filtered_rmw_implementation=None, exit_on_match=True
+        filtered_prefixes=None, filtered_patterns=None, filtered_rmw_implementation=None,
+        exit_on_match=True,
     ):
         super(LineOutput, self).__init__()
         if filtered_prefixes is None:
             self.filtered_prefixes = get_default_filtered_prefixes()
         else:
             self.filtered_prefixes = filtered_prefixes
+        if filtered_patterns is None:
+            self.filtered_patterns = get_default_filtered_patterns()
+        else:
+            self.filtered_patterns = filtered_patterns
 
         if filtered_rmw_implementation:
             rmw_output_filter = get_rmw_output_filter(filtered_rmw_implementation)
-            self.filtered_prefixes.extend(rmw_output_filter)
+            self.filtered_patterns.extend(rmw_output_filter)
```

Comment:
Why should the rmw filters need to be patterns? If a prefix filter is sufficient it would be good if an rmw impl. could use that.

Reply:
I understood from your comment above that, while we want users to give users the choice between patterns or prefixes, it's not necessary for "internal" rmw filters:

> Updating the rmw implementations to always provide patterns seems fine since that won't affect users (except those implementing their own rmw impl. with custom filters).

Reply:
That was 20 days ago -  a lot has happened since then :wink:

Sure, we could make all rmw impl. use regex since we kind of maintain most of them. That was the main rational in the first review. But my latest comment was about efficiency - why use a regex if a prefix does the job just fine? I haven't spend any thought on how this could be done. An rmw impl. would need to somehow declare the type of their filters - either by using a different resource key or embedding the type information in the resource. The first seems to be easier.